workflow/Snakefile

```diff
@@ -47,6 +51,35 @@ BIOSAMPLE_ACTIVITES = {
 # defined to be compatible with training workflow, which can start from an existing ABC directory
 ABC_BIOSAMPLES_DIR = {biosample: os.path.join(RESULTS_DIR, biosample) for biosample in BIOSAMPLE_DF.biosample}
 
+# add model directory path to biosamples df
```

Comment:
Some cleaner code suggestions:
1. We'd want this file to stay simple. It's better to move this logic into the utils.smk file. And in this file, we can just call something like `expand_biosamples_df(BIOSAMPLE_DF)`

2. We can combine all the logic (line 54-81) into 1 function that looks like the following:
```
def expand_biosample_df(biosample_df):
  new_rows = []
  for row in biosample_df:
    if model_dir not in row:
        row['model_dir'] = infer model_dir
    
    for model_dir in row['model_dir'].split(','):
      new_row = row.copy()
      new_row['model_dir'] = model_dir
      new_row['model_base'] = ..
      new_row['threshold'] = ..
      new_rows.append(new_row)
  return pd.DataFrame(new_rows)  
```